tasks/main.yml

```diff
@@ -44,7 +44,8 @@
     service: "{{ item.service | default(omit) }}"
     port: "{{ item.port | default(omit) }}"
     source_port: "{{ item.source_port | default(omit) }}"
-    forward_port: "{{ item.forward_port | default(omit) }}"
+    forward_port: "{{ (item.forward_port is mapping) |
+      ternary([item.forward_port], item.forward_port) | default(omit) }}"
```

Comment:
I don't think you need to change this - the type of `item.forward_port` should always be a `list` https://github.com/linux-system-roles/firewall/blob/6521366e9ef72492ff7563fb5c5073ea375796d1/library/firewall_lib.py#L428 and

Reply:
Oh, should we not allow support for something like:
```yaml
firewall:
  forward_port:
    port: 80
    proto: tcp
```
(When "-" is not placed before the port being forwarded)

Reply:
ok - I see - https://github.com/linux-system-roles/firewall#forward_port "dict or list of dicts in the format like ansible.posix.firewalld"
Then I think the type handling should be done in the module, not here - I think the logic is going to get complicated, because the value of `forward_port` could be a single string, a list of strings, a single dict, or a list of dict - or even a list of mixed dict and string
But that means we'll have to change https://github.com/linux-system-roles/firewall/blob/6521366e9ef72492ff7563fb5c5073ea375796d1/library/firewall_lib.py#L428 to also be `raw` . . . on the other hand, every other similar parameter says "str or list" in the docs e.g. https://github.com/linux-system-roles/firewall#port - but they should all fail Ansible argument validation because they all take a `list` e.g. https://github.com/linux-system-roles/firewall/blob/6521366e9ef72492ff7563fb5c5073ea375796d1/library/firewall_lib.py#L424 - I'm assuming if you try to pass a string instead of a list, you will get an error - so really the documentation is wrong, or the doc writer intended to support both `str` and `list` (somehow - by writing custom Ansible argument validation code?)

I guess what I'm saying is that we should just assume the value of `forward_port` here is a `list`

Then, I guess we can have a separate discussion about what to do about the README
* Change it to reflect the actual code - only `list` is accepted
* Change the code to reflect the README - which means implementing our own argument validation

Reply:
I believe they support a string value for `port`, they ansible just appears to pass strings in as `[string]`, which works for a single port. The issue is when a `dict` is passed in to a list and ansible-playbook does not seem to know what to do to convert it to a list like it does with `str`

Nonetheless, I do find using our own argument validation to be largely unnecessary if there is no issue with assuming a list is passed in. I'll close this conversation and reopen another over README

Reply:
What I'm going to do for the time being is update README to match the actual functionality that is present. This would just mean removing the mention of single dict in the `forward-port` section.

Reply:
Update, re-implemented single dict with minimal changes in firewall_lib.py